File: challenge-02/main.dart
import 'dart:io';

void main() {
  File('challenge-02/data.txt').readAsString().then((String data) {
    String result = '';
    data.split('').fold(0, (prev, next) {
      switch (next) {
        case '#':
          return prev + 1;
        case '@':
          return prev - 1;
        case '*':
          return prev * prev;
        case '&':
          result += prev.toString();
      }
      return prev;
    });
    print(result);
  }).onError((_, __) {
    print('archivo inexistente');
  });
}

File: challenge-02/main.py

response = ''
with open('challenge-02/data.txt','r') as f:
    prev = 0
    for word in list(f.readline()):
        if(word == '#'):
            prev += 1
            continue
        if(word == '@'):
            prev -= 1
            continue
        if(word == '*'):
            prev *= prev
            continue
        if(word == '&'):
            response += str(prev)
print (response)


